modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -673,15 +675,18 @@ private CompletableFuture<?> updateAssignmentInternal(ConfigurationNotificationE
                 return failedFuture(e);
             }
 
+            TableImpl table = tablesById.get(tblId);
+
+            // TODO: IGNITE-19082 Need another way to wait for indexes
+            table.addIndexesToWait(collectTableIndexes(tablesCfg.value(), tblId));
```

Comment:
Why is this code located in `onUpdateAssignments` and not in table or index creation listeners?

Reply:
Why is this code even needed? I can't understand this from the ticket description and we already populate in index-to-wait map when registering indices

Reply:
Before recovery raft groups, it is important for us to know the list of indices that will be updated on recovery (in `PartitionListener`), now we are waiting only for the PK index, and we forget about the rest, which is not correct.

Reply:
After an in-person discussion we decided that this code should actually be moved to the table creation listeners

Reply:
Discussed personally, I think that the problem has become clearer.